test/topology_custom/test_multidc.py

```diff
@@ -0,0 +1,209 @@
+@pytest.mark.asyncio
+@pytest.mark.skip(reason="Too heavy for CI, but OK for local run")
+async def test_multidc(request: pytest.FixtureRequest, manager: ManagerClient) -> None:
+    logger.info("Creating a new node")
+    for i in range(30):
+        s_info = await manager.server_add(
+            config=CONFIG,
+            property_file={'dc': f'dc{i}', 'rack': 'myrack1'}
+        )
+        logger.info(s_info)
+    random_tables = RandomTables(request.node.name, manager, unique_name(), 3)
+    logger.info("Creating new tables")
+    await random_tables.add_tables(ntables=3, ncolumns=3)
+    await random_tables.verify_schema()
+
+
+@pytest.mark.asyncio
+async def test_putget_2dc_rf1(
+        request: pytest.FixtureRequest, manager: ManagerClient
+) -> None:
+    """Simple put-get test for 2 DC with one node each (RF=1) [catches #3539]"""
+    ks = "test_ks"
+    cf = "test_cf"
+    table_name = "test_table_name"
+    columns = [Column("name", TextType), Column("value", TextType)]
+    logger.info("Create two servers in different DC's")
+    for i in [1, 2]:
+        s_info = await manager.server_add(
+            config=CONFIG,
+            property_file={"dc": f"dc{i}", "rack": "myrack"},
+        )
+        logger.info(s_info)
+    conn = manager.get_cql()
+    random_tables = RandomTables(request.node.name, manager, ks, 1)
+    logger.info("Add table")
+    await random_tables.add_table(ncolumns=2, columns=columns, pks=1, name=table_name)
+    conn.execute(f"USE {ks}")
+    conn.execute(
+        f"CREATE COLUMNFAMILY {cf} ( key varchar, c varchar, v varchar, PRIMARY KEY(key, c)) WITH comment='test cf'"
+    )
+
+    logger.info("Perform insert/overwrite")
+    putget(conn, cf, ks)
+
+
+@pytest.mark.asyncio
+async def test_putget_2dc_rf2(
+        request: pytest.FixtureRequest, manager: ManagerClient
+) -> None:
+    """Simple put-get test for 2 DC with 2 node each (RF=2) -- tests cross-DC efficient writes"""
+    ks = "test_ks"
+    cf = "test_cf"
+    table_name = "test_table_name"
+    columns = [Column("name", TextType), Column("value", TextType)]
+    logger.info("Create two servers in different DC's")
+    for i in [1, 1, 2, 2]:
+        s_info = await manager.server_add(
+            config=CONFIG,
+            property_file={"dc": f"dc{i}", "rack": "myrack"},
+        )
+        logger.info(s_info)
+    conn = manager.get_cql()
+    logger.info(conn.execute('desc cluster').one())
+    random_tables = RandomTables(request.node.name, manager, ks, 2)
+    await random_tables.add_table(ncolumns=2, columns=columns, pks=1, name=table_name)
+    conn.execute(f"USE {ks}")
+    conn.execute(
+        f"CREATE COLUMNFAMILY {cf} ( key varchar, c varchar, v varchar, PRIMARY KEY(key, c)) WITH comment='test cf'"
+    )
+
+    logger.info("Perform insert/overwrite")
+    putget(conn, cf, ks)
+
+
+@pytest.mark.asyncio
+async def test_query_dc_with_rf_0_does_not_crash_db(request: pytest.FixtureRequest, manager: ManagerClient):
```

Comment:
Another thing you forgot to mention in the commit message is what tests you are adding, and why. This particular test seems a direct "borrowing" of the existing dtest multidc_additional_tests.py. I think it was worth mentioning. And it may be worth considering (please do!) deleting this dtest later - we don't need exactly the same test twice.

Are other tests added here also translations of existing dtests?

Reply:
All tests presented here are borrowed: 3 from dtests and 1 from https://github.com/kostja/scylla/blob/multidc2/test/topology_custom/test_multidc.py
Sure, I'll delete this tests from dtest once this PR will be merged. You are completely right there is no need to have the same tests in different repos.